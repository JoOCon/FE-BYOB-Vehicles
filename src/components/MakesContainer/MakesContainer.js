import React from 'react';
import MakeCard from '../MakeCard/MakeCard';
import './MakesContainer.css';

const MakesContainer = (props) => {
  const { makes } = props;
  const makesList = makes.map(make => (
    <MakeCard
      {...make}
      key={make.id}
    />
  ));
  return (
    <section className="makes-cont">
      {makesList}
    </section>
  );
};

export default MakesContainer;
