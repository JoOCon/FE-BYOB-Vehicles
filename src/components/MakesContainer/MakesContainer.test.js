import React from 'react';
import { shallow } from 'enzyme';
import MakesContainer from './MakesContainer';

describe('MakesContainer', () => {
  it('should match snapshot with makes', () => {
    const mockMakes = [{ name: 'garry' }, { name: 'billy' }];
    const wrapper = shallow(
      <MakesContainer
        makes={mockMakes}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
