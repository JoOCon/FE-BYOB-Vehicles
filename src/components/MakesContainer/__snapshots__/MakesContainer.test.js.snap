// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakesContainer should match snapshot with makes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MakesContainer
    makes={
      Array [
        Object {
          "name": "garry",
        },
        Object {
          "name": "billy",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <UNDEFINED
          name="garry"
        />,
        <UNDEFINED
          name="billy"
        />,
      ],
      "className": "makes-cont",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "garry",
        },
        "ref": null,
        "rendered": null,
        "type": Object {},
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "billy",
        },
        "ref": null,
        "rendered": null,
        "type": Object {},
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <UNDEFINED
            name="garry"
          />,
          <UNDEFINED
            name="billy"
          />,
        ],
        "className": "makes-cont",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "garry",
          },
          "ref": null,
          "rendered": null,
          "type": Object {},
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "billy",
          },
          "ref": null,
          "rendered": null,
          "type": Object {},
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
